DataFormats/ParticleFlowReco/interface/PFBlockElement.h

```diff
@@ -111,12 +111,25 @@ namespace reco {
     virtual bool isLinkedToDisplacedVertex() const { return false; }
 
     // Glowinski & Gouzevitch
-    void setMultilinks(const PFMultiLinksTC& ml) { multilinks_ = ml; }
-    void setIsValidMultilinks(bool isVal) { multilinks_.isValid = isVal; }
-    void setMultilinksList(const PFMultilinksType& links) { multilinks_.linkedClusters = links; }
-
-    bool isMultilinksValide() const { return multilinks_.isValid; }
-    const PFMultilinksType& getMultilinks() const { return multilinks_.linkedClusters; }
+    void setMultilinks(const PFMultiLinksTC& ml, Type type) { multilinks_[type] = ml; }
+    void setIsValidMultilinks(bool isVal, Type type) { multilinks_[type].isValid = isVal; }
+
+    bool isMultilinksValide(Type type) const {
+      auto it = multilinks_.find(type);
+      if (it != multilinks_.end())
+        return it->second.isValid;
+      else
+        return false;  // no multilinks_ for the specified type
+    }
+    const PFMultilinksType& getMultilinks(Type type) const {
+      auto it = multilinks_.find(type);
+      if (it != multilinks_.end())
+        return it->second.linkedClusters;
+      else {
+        const static reco::PFMultilinksType nullPFMultiLinksTC;
+        return nullPFMultiLinksTC;  // no multilinks_ for the specified type, so return null
```

Comment:
Returning a reference that points to a static object with default initialization... Perhaps it would be better to throw an exception, rather than introduce a custom "null type" ?

Reply:
never, NEVER, throw to report an error

Reply:
thanks for your input - I guess this is something that's prevalent in PF code [1] and I was going based off of that. What would be your suggestion here? Anyway, introducing a custom null type just to satisfy the function signature seems odd. My understanding is that one does not expect this particular case to happen in normal operation?

[1] e.g. https://github.com/cms-sw/cmssw/blob/ec4c535880057c2e4d4d8b603f872bc9ea9eba43/RecoParticleFlow/PFProducer/plugins/linkers/HCALAndHOLinker.cc#L35

Reply:
Just to clarify based on the mattermost discussion [1], @hatakeyamak do you expect to reach this point in the code at any time? If not, it might be better to throw an assert (as opposed to an exception, which can be catchable) to signal that something went wrong.

[1] https://mattermost.web.cern.ch/cms-o-and-c/pl/eyg7chonaifoz81h693kono1zw

Reply:
I saw the discussion. I think this shouldn't happen. Before we check the multilinks, it should be filled. Let us check that with some local tests and jenkins.

Reply:
> never, NEVER, throw to report an error

if the alternative is an assert, as implied in https://github.com/cms-sw/cmssw/pull/31295#discussion_r484881272, I'd still rather have an exception.
Neither can imply that the program execution can just continue though.

Reply:
Thanks. Ken, let us know if you have any objection to introducing it.

Reply:
So I think I will do exception.

Reply:
Maybe this whole function could even be replaced with `multilinks_.at(type)`, as the function `getMultilinks` doesn't do anything else.

Reply:
@jpata Actually, there are cases that multilinks_ are not filled, because in some ecases there are requirements for, for examples, tracks to be considered for links in HCAL (make sure they reach the HCAL surface).  But getMultilinks is called only when isMultilinksValide is good, so I think your suggestion works. I changed as suggested.